### PR TITLE
Setup Virtual Session drawer

### PR DESCRIPTION
**Pull request by djnunez-aot**
*Issue #1182 :*
https://github.com/bcgov/met-public/issues/1182

-Add Virtual session drawer

-Create info paper for virtual event

-Update event context

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        try {
+            setIsCreating(true);
+            const { description, session_link, session_link_text, date, time_from, time_to } = validatedData;
+            const time_from_split = time_from.split(':');
+            const time_to_split = time_to.split(':');
+            const dateFrom = dayjs(date)
+                .set('hour', Number(time_from_split[0]))
+                .set('minute', Number(time_from_split[1]));
+            const dateTo = dayjs(date).set('hour', Number(time_to_split[0])).set('minute', Number(time_to_split[1]));
+            console.log('POST EVENT!!');
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx:62]: console.log

File: met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        console.log('ON SUBMIT!!!!');
+        if (!widget) {
+            return;
+        }
+        const validatedData = await schema.validate(data);
+        try {
+            setIsCreating(true);
+            const { description, session_link, session_link_text, date, time_from, time_to } = validatedData;
+            const time_from_split = time_from.split(':');
+            const time_to_split = time_to.split(':');
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx:57]: I had a mind to cleanup this section in the other drawer. Just to export them into a separate function called something like getDates that returns { dateFrom: ..., dateTo: ...}.

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
Console.log left, and a suggestion

File: met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx (hunk 2 above)
Comment (djnunez-aot) [reply met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx:57]: function added to date helper

Comment (jadmsaadaot) [reply met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx:55]: Yeah I see it has been cleaned up in add in person event, could be used here too no?

Comment (jadmsaadaot) [reply met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx:55]: yeah, could we please add the time split there too?

Comment (jadmsaadaot) [reply met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx:55]: It's not a very common function, so I would create a a utils file for it under the Events folder. But okay to also keep it in common

Comment (djnunez-aot) [reply met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx]: Added a utils file

**Review by VineetBala-AOT — APPROVED** [review]

File: met-web/src/components/engagement/form/EngagementWidgets/Events/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import dayjs from 'dayjs';
+
+export const getDates = (date: string, time_from: string, time_to: string) => {
+    const time_from_split = time_from.split(':');
+    const time_to_split = time_to.split(':');
+    const dateFrom = dayjs(date).set('hour', Number(time_from_split[0])).set('minute', Number(time_from_split[1]));
+    const dateTo = dayjs(date).set('hour', Number(time_to_split[0])).set('minute', Number(time_to_split[1]));
+
+    return { dateFrom: dateFrom, dateTo: dateTo };
+};
```
Comment (saravanpa-aot) [inline met-web/src/components/engagement/form/EngagementWidgets/Events/utils.tsx:10]: can you add some comments here?
getDates is not very clear on what it does ; so may be rename it 
also please be very liberal on writing comments

Comment (djnunez-aot) [reply met-web/src/components/engagement/form/EngagementWidgets/Events/utils.tsx]: Added some comments and changed the name

**Review by jadmsaadaot — APPROVED** [review]

**Pull request merged**